packages/core/core/src/public/BundleGraph.js

```diff
@@ -11,47 +11,50 @@ import type {
   Symbol,
   SymbolResolution,
 } from '@parcel/types';
-import type {ParcelOptions} from '../types';
+import type {Bundle as InternalBundle, ParcelOptions} from '../types';
 import type InternalBundleGraph from '../BundleGraph';
 
 import nullthrows from 'nullthrows';
-import {DefaultWeakMap} from '@parcel/utils';
 
 import {assetFromValue, assetToAssetValue} from './Asset';
-import {Bundle, bundleToInternalBundle} from './Bundle';
+import {bundleToInternalBundle} from './Bundle';
 import Dependency, {dependencyToInternalDependency} from './Dependency';
 import {mapVisitor} from '../Graph';
 
-const internalBundleGraphToBundleGraph: DefaultWeakMap<
-  ParcelOptions,
-  WeakMap<InternalBundleGraph, BundleGraph>,
-> = new DefaultWeakMap(() => new WeakMap());
 // Friendly access for other modules within this package that need access
 // to the internal bundle.
 const _bundleGraphToInternalBundleGraph: WeakMap<
-  IBundleGraph,
+  IBundleGraph<IBundle>,
   InternalBundleGraph,
 > = new WeakMap();
 export function bundleGraphToInternalBundleGraph(
-  bundleGraph: IBundleGraph,
+  bundleGraph: IBundleGraph<IBundle>,
 ): InternalBundleGraph {
   return nullthrows(_bundleGraphToInternalBundleGraph.get(bundleGraph));
 }
 
-export default class BundleGraph implements IBundleGraph {
+type BundleFactory<TBundle: IBundle> = (
+  InternalBundle,
+  InternalBundleGraph,
+  ParcelOptions,
+) => TBundle;
+
+export default class BundleGraph<TBundle: IBundle>
+  implements IBundleGraph<TBundle> {
   #graph; // InternalBundleGraph
   #options; // ParcelOptions
+  _createBundle: BundleFactory<TBundle>;
```

Comment:
This doesn't seem to work as a private property (`#createBundle`) with Flow. The constructor is still accessible through the `constructor` property of the wrapper objects externally anyway.

Reply:
Why doesn't it work?

Reply:
A bug in flow, private properties are not callable (though assigning them beforehand works): https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBjGBDAzrsAcTAG9UwwBiKOOAbnLEYzgDtcAXAJwFcMO4XABQBKMhQocAFgEtcAOmq0wAXjCjVAPjDceAUwYUAvugoAjLnADWe1hvESuejjy6sdshUrijDYE4wIglbYljysACb2jBQwzmAAHqoecoo09DFgTi5uib6MJkZAA